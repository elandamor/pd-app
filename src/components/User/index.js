/**
 * User
 * @type {Component} Presentational
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
// Components
import Avatar from '../Avatar';

const Wrapper = styled.div`
  align-items: center;
  display: flex;
`;

const Info = styled.div`
  display: flex;
  flex: 1;
  flex-direction: column;
  margin: 0 auto;
  overflow: hidden;
  padding: 6px 0;
  text-decoration: none;

  span {
    display: block;
    font-size: 11px;
    color: ${(props) => props.theme.isDark ? '#fff' : '#000'};
    letter-spacing: 0.01rem;
    line-height: 1.5;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;

    &.name {
      font-size: 13px;
      font-weight: 900;
      letter-spacing: 0.01rem;
      margin-bottom: -2px;
    }

    &.username {
      color: ${(props) => props.theme.isDark ? '#D3D4D8' : '#555'};
      font-weight: 400;
    }
  }
`;

const User = ({ avatar, className, name, username }) => { // eslint-disable-line arrow-body-style
  return (
    <Wrapper className={`c-user ${className || ''}`}>
      <Avatar src={avatar} />
      <Info className="c-info">
        <span className="name">{name}</span>
        <span className="username">{`@${username}`}</span>
      </Info>
    </Wrapper>
  );
};

User.propTypes = {
  avatar: PropTypes.string,
  className: PropTypes.string,
  name: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
};

export default User;
