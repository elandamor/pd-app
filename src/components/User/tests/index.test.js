import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';
import 'jest-enzyme';

import User from '../index';

const themeLight = {
  palette: {
    cardBorderColor: '#ededed',
  },
};

describe('User component renders', () => {
  it('correctly', () => {
    const rendered = renderer.create(
      <ThemeProvider theme={themeLight} >
        <User
          avatar="image.jpg"
          name="Thandolwethu Mpofu"
          username="elandamor"
        />
      </ThemeProvider>
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});

