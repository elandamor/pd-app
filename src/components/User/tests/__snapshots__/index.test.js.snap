// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component renders correctly 1`] = `
<div
  className="c-user  foCcDl"
>
  <div
    className="c-avatar  kGqdWm"
  >
    <img
      className="c-image lazy-image gGwVwM"
      data-src="image.jpg"
      src="image.jpg"
    />
  </div>
  <div
    className="c-info eEhkFR"
  >
    <span
      className="name"
    >
      Thandolwethu Mpofu
    </span>
    <span
      className="username"
    >
      @elandamor
    </span>
  </div>
</div>
`;
